File: doc-examples/example-java/src/main/java/example/Person.java
package example;

import com.fasterxml.jackson.annotation.JsonFilter;
import io.micronaut.serde.annotation.Serdeable;

@Serdeable
@JsonFilter("person-filter") // <1>
public class Person {
    private final String name;
    private final String preferredName;

    public Person(String name, String preferredName) {
        this.name = name;
        this.preferredName = preferredName;
    }

    public String getName() {
        return name;
    }

    public String getPreferredName() {
        return preferredName;
    }
}

File: doc-examples/example-python/src/main/python/example/Person.py
from dataclasses import dataclass

from com.fasterxml.jackson.annotation import JsonFilter
from micronaut.serde.annotation import Serdeable


@Serdeable
@JsonFilter("person-filter")  # <1>
@dataclass
class Person:
    name: str
    preferredName: str | None = None
